lib/maths/unittest/CBoostedTreeTest.cc

```diff
@@ -1221,6 +1222,76 @@ BOOST_AUTO_TEST_CASE(testEstimateMemoryUsedByTrain) {
     }
 }
 
+BOOST_AUTO_TEST_CASE(testEstimateMemoryUsedByTrainWithTestRows) {
+
+    // Test estimation of the memory used training a model.
+
+    test::CRandomNumbers rng;
+
+    std::size_t rows{1000};
+    std::size_t cols{6};
+    std::size_t capacity{600};
+    std::int64_t previousEstimatedMemory{LLONG_MAX};
```

Comment:
nit: is the C++ way of doing this
```suggestion
    std::int64_t previousEstimatedMemory{std::numeric_limits<std::uint64_t>::max()};
```